src/lib/hasher/hasher.ts

```diff
@@ -168,14 +174,82 @@ export function getLeafHasher<V extends CCIPVersion = CCIPVersion>({
   destChainSelector,
   onRamp,
   version,
-}: Lane<V>): LeafHasher {
+}: Lane<V>): LeafHasher<V> {
   switch (version) {
     case CCIPVersion.V1_2:
     case CCIPVersion.V1_5:
-      return getV12LeafHasher(sourceChainSelector, destChainSelector, onRamp)
+      return getV12LeafHasher(sourceChainSelector, destChainSelector, onRamp) as LeafHasher<V>
     case CCIPVersion.V1_6:
-      return getV16LeafHasher(sourceChainSelector, destChainSelector, onRamp)
+      return getV16LeafHasher(sourceChainSelector, destChainSelector, onRamp) as LeafHasher<V>
     default:
       throw new Error(`Unsupported CCIP version: ${version}`)
   }
 }
+
+export type AptosLeafHasher = (message: AptosCCIPMessage) => string
+
+export const makeAptosHasher =
+  (sourceChainSelector: bigint, destChainSelector: bigint, onRamp: string): AptosLeafHasher =>
+  (message: AptosCCIPMessage): string =>
+    hashAptosMessage(message, hashAptosMetadata(sourceChainSelector, destChainSelector, onRamp))
+
+const encode = (target: string, value: string | bigint | number) =>
+  defaultAbiCoder.encode([target], [value])
+
+/**
+ * Encodes dynamic bytes without the extra length prefix.
+ * The default ABI encoding for type "bytes" is: 32-byte length prefix + the actual data padded.
+ * This helper strips the 32-byte length prefix, mimicking the Move implementation.
```

Comment:
EVM also has something similar, but this is usually taken care by the abi decoder;
i.e. the abi.encoded words also have the length in first word, then the rest of the array; but when abi decoding (e.g. from ethers), we get only the bytearray that we expect, as the client/parser takes care of this padding.

Shouldn't we expect Aptos client/decoder to also do this for us?